spec/bitcoin_ticker_spec.rb

```diff
@@ -93,21 +101,21 @@ module BitcoinTicker
   end
 
   describe SlackNotifier do
-    let(:slack_notifier) { described_class.new }
     let(:current_price) { 123.45 }
     let(:webhook_url) { "https://slack.com/webhook_url/123" }
-    let(:text) { "Bitcoin is up to $#{current_price}" }
-    let(:username) { "bitcoin-ticker" }
-    let(:icon_url) { "https://en.bitcoin.it/w/images/en/2/29/BC_Logo_.png" }
-    let(:payload) { "{\"text\":\"#{text}\",\"username\":\"#{username}\",\"icon_url\":\"#{icon_url}\"}" }
-    before { expect(Net::HTTP).to receive(:post_form).with URI(webhook_url), payload: payload }
+    let(:slack_notifier) { described_class.new webhook_url: webhook_url }
+
     it "posts current price to slack" do
-      slack_notifier.notify webhook_url, current_price, true
+      expect(slack_notifier).to receive(:http_call)
+
+      slack_notifier.notify current_price, true
```

Comment:
this would be an integration test, and I'd be observing the interaction with something like VCR (real blackbox testing).